cmd/describe/machinepool/nodepool.go

```diff
@@ -0,0 +1,115 @@
+package machinepool
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	ocmOutput "github.com/openshift/rosa/pkg/ocm/output"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+func describeNodePool(r *rosa.Runtime, cluster *cmv1.Cluster, clusterKey string, nodePoolID string) error {
+	r.Reporter.Debugf("Fetching node pool '%s' for cluster '%s'", nodePoolID, clusterKey)
+	nodePool, exists, err := r.OCMClient.GetNodePool(cluster.ID(), nodePoolID)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		r.Reporter.Infof("Machine pool '%s' not found", nodePoolID)
+		return nil
+	}
+
+	_, scheduledUpgrade, err := r.OCMClient.GetHypershiftNodePoolUpgrade(cluster.ID(), clusterKey, nodePoolID)
+	if err != nil {
+		return err
+	}
+
+	if output.HasFlag() {
+		var formattedOutput map[string]interface{}
+		formattedOutput, err = formatNodePoolOutput(nodePool, scheduledUpgrade)
+		if err != nil {
+			return err
+		}
+		err = output.Print(formattedOutput)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// Prepare string
+	nodePoolOutput := fmt.Sprintf("\n"+
+		"ID:                         %s\n"+
+		"Cluster ID:                 %s\n"+
+		"Autoscaling:                %s\n"+
+		"Desired replicas:           %s\n"+
+		"Current replicas:           %s\n"+
+		"Instance type:              %s\n"+
+		"Labels:                     %s\n"+
+		"Taints:                     %s\n"+
+		"Availability zone:          %s\n"+
+		"Subnet:                     %s\n"+
+		"Version:                    %s\n"+
+		"Autorepair:                 %s\n"+
+		"Tuning configs:             %s\n"+
+		"Message:                    %s\n",
+		nodePool.ID(),
+		cluster.ID(),
+		ocmOutput.PrintNodePoolAutoscaling(nodePool.Autoscaling()),
+		ocmOutput.PrintNodePoolReplicas(nodePool.Autoscaling(), nodePool.Replicas()),
+		ocmOutput.PrintNodePoolCurrentReplicas(nodePool.Status()),
+		ocmOutput.PrintNodePoolInstanceType(nodePool.AWSNodePool()),
+		ocmOutput.PrintLabels(nodePool.Labels()),
+		ocmOutput.PrintTaints(nodePool.Taints()),
+		nodePool.AvailabilityZone(),
+		nodePool.Subnet(),
+		ocmOutput.PrintNodePoolVersion(nodePool.Version()),
+		ocmOutput.PrintNodePoolAutorepair(nodePool.AutoRepair()),
+		ocmOutput.PrintNodePoolTuningConfigs(nodePool.TuningConfigs()),
+		ocmOutput.PrintNodePoolMessage(nodePool.Status()),
+	)
+
+	// Print scheduled upgrades if existing
+	if scheduledUpgrade != nil {
```

Comment:
I'd review if the others "describe" commands have also conditional rendering.

Just to use the same pattern (show always the info with empty spaces vs only show it when is present).

Reply:
yes, this is the same pattern we are using in the `describe cluster`, see https://github.com/openshift/rosa/blob/master/cmd/describe/cluster/cmd.go#L363-L369